web-app/src/games/games.service.ts

```diff
@@ -53,60 +52,51 @@ export class GamesService {
         return this.gamesRepository.findOne(id);
     }
 
-    async createNewGame(size: number, player1SessionId: string): Promise<Game> {
+    async createNewGame(size: number, player1Username: string): Promise<Game> {
         if (!size) {
             size = DEFAULT_BOARD_SIZE;
         } else {
             size = +size;
         }
-        const player1 = this.usersRepository.create({
-            lastSessionId: player1SessionId,
-            username: player1SessionId
-        });
+        const player1 = await this.usersService.getOrCreate(player1Username);
         const game = this.gamesRepository.create({
             state: initNewGameState(size),
-            player1: await this.usersRepository.save(player1),
+            player1: player1,
         });
         return await this.gamesRepository.save(game);
     }
 
-    async createNewGameFromFile(player1SessionId: string): Promise<Game> {
-        const player1 = this.usersRepository.create({
-            lastSessionId: player1SessionId,
-            username: player1SessionId
-        });
+    async createNewGameFromFile(player1Username: string): Promise<Game> {
+        const player1 = await this.usersService.getOrCreate(player1Username);
         const game = this.gamesRepository.create({
             state: this.getBoardStateFromFile(),
-            player1: await this.usersRepository.save(player1),
+            player1: player1,
         });
         return await this.gamesRepository.save(game);
     }
 
-    async joinGame(gameId: number, player2SessionId: string): Promise<Game> {
+    async joinGame(gameId: number, player2Username: string): Promise<Game> {
         const game = await this.gamesRepository.findOne(gameId);
         if (!game.player2) {
-            if (player2SessionId !== game.player1.lastSessionId) {
-                const player2 = this.usersRepository.create({
-                    lastSessionId: player2SessionId,
-                    username: player2SessionId
-                });
-                game.player2 = await this.usersRepository.save(player2);
+            if (player2Username !== game.player1.username) {
+                const player2 = await this.usersService.getOrCreate(player2Username);
+                game.player2 = player2;
                 return await this.gamesRepository.save(game);
             } else {
-                throw Error(`Player2 (sessionId: ${player2SessionId}) cannot be the same as player1 (sessionId: ${game.player1.lastSessionId})`);
+                throw Error(`Player2 cannot be the same user than player1`);
             }
         } else {
             throw Error(`Game ${gameId} already has a player2 (username: ${game.player2.username})`);
         }
     }
 
-    getGameAndStatus(game: Game, playerSessionId: string): GameAndStatus {
+    getGameAndStatus(game: Game, playerName: string): GameAndStatus {
```

Comment:
```suggestion
    getGameAndStatus(game: Game, playerUserName: string): GameAndStatus {
```
Naming would be more consistant this way 🙂